modules/gapi/test/s11n/gapi_s11n_tests.cpp

```diff
@@ -614,6 +638,87 @@ TEST_F(S11N_Basic, Test_RunArg_RMat_Scalar_Mat) {
     EXPECT_EQ(0, cv::norm(mat, out_mat));
 }
 
+TEST_F(S11N_Basic, Test_RunArg_MediaFrame) {
+    cv::Mat mat = cv::Mat::eye(cv::Size(128, 64), CV_8UC3);
+    auto frame = cv::MediaFrame::Create<MyMediaFrameAdapter>(mat, 42, "It actually works");
+    auto v = cv::GRunArgs{ cv::GRunArg{ frame } };
+
+    const std::vector<char> sargsin = cv::gapi::serialize(v);
+    cv::GRunArgs out = cv::gapi::deserialize<cv::GRunArgs, MyMediaFrameAdapter>(sargsin);
+    cv::MediaFrame out_mat = cv::util::get<cv::MediaFrame>(out[0]);
+    auto adapter = out_mat.get<MyMediaFrameAdapter>();
+    EXPECT_EQ(42, adapter->getVal());
+    EXPECT_EQ("It actually works", adapter->getStr());
+}
+
+TEST_F(S11N_Basic, Test_RunArg_MediaFrame_Scalar_Mat) {
+    cv::Mat mat = cv::Mat::eye(cv::Size(128, 64), CV_8UC3);
+    auto frame = cv::MediaFrame::Create<MyMediaFrameAdapter>(mat, 42, "It actually works");
+    cv::Scalar sc(111);
+    auto v = cv::GRunArgs{ cv::GRunArg{ frame }, cv::GRunArg{ sc }, cv::GRunArg{ mat } };
+
+    const std::vector<char> sargsin = cv::gapi::serialize(v);
+    cv::GRunArgs out = cv::gapi::deserialize<cv::GRunArgs, MyMediaFrameAdapter>(sargsin);
+    cv::MediaFrame out_frame = cv::util::get<cv::MediaFrame>(out[0]);
+    auto adapter = out_frame.get<MyMediaFrameAdapter>();
+    EXPECT_EQ(42, adapter->getVal());
+    EXPECT_EQ("It actually works", adapter->getStr());
+
+    cv::Scalar out_sc = cv::util::get<cv::Scalar>(out[1]);
+    EXPECT_EQ(sc, out_sc);
+
+    cv::Mat out_mat = cv::util::get<cv::Mat>(out[2]);
+    EXPECT_EQ(0, cv::norm(mat, out_mat));
```

Comment:
For such check I'd add some initialization for mat, don't you think so?

Reply:
I just copypasted it from another test. There are like 20 places here without mat initialization, let's do it separately

Reply:
Oh, it's actually initialized - it's an eye matrix (`cv::Mat::eye()`)